infra/modules/database-serverless/main.tf

```diff
@@ -0,0 +1,204 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+data "aws_vpc" "default" {
+  default = true
+}
+module "random_admin_database_password" {
+  source = "../random-password"
+  # Mysql password is maxed out at 41 chars
+  length = var.database_type == "mysql" ? 41 : 48
+}
+
+locals {
+  admin_user                 = "app_usr"
+  admin_user_secret_name     = "/metadata/db/${var.database_name}-admin-user"
+  admin_password_secret_name = "/metadata/db/${var.database_name}-admin-password"
+  admin_db_url_secret_name   = "/metadata/db/${var.database_name}-admin-db-url"
+  admin_db_host_secret_name  = "/metadata/db/${var.database_name}-admin-db-host"
+  database_name_formatted    = replace("${var.database_name}", "-", "_")
+  admin_password             = var.admin_password == "" ? module.random_admin_database_password.random_password : var.admin_password
+}
+
+############################
+## Security Groups ##
+############################
+
+resource "aws_security_group" "database" {
+  description = "Allow inbound TCP access to database port"
+  name        = "${var.database_name}-database-rds"
+  vpc_id      = data.aws_vpc.default.id
+
+  ingress {
+    from_port   = var.database_port
+    to_port     = var.database_port
+    protocol    = "tcp"
+    cidr_blocks = [data.aws_vpc.default.cidr_block]
+    description = "Allow inbound TCP access to database port"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+############################
+## Database Configuration ##
+############################
+resource "aws_db_instance" "database" {
+  # checkov:skip=CKV_AWS_157:Multi-AZ is mostly unessecary for a project of this size.
+  identifier        = var.database_name
+  allocated_storage = 20
```

Comment:
Please verify:
- that this is the default recommended storage 
- AND that it's sufficient for both of our applications